docs: Add new Getting Started page (#2638)
MystenLabs/walrus: docs/book/usage/started.md

# Getting Started with Walrus

Walrus is a platform for building efficient and resilient data markets, where data is stored as

> *blobs*. Walrus stores all blobs as an array of bytes with a fixed size, and you can store any

Reviewer: Important: Not quite sure about the "fixed size". Blobs can have arbitrary sizes, although they may be padded to adhere to alignment requirements.

*network*.

### Available networks

Sui and Walrus each have the following available networks:

- **Testnet** is a sandbox-like network where you can receive test tokens for free to use for the
  network's fees. You can build, test, and debug software packages on Testnet. Testnet does not
  guarantee data persistence and might wipe data at any time without warning.
- **Mainnet** is a production environment where you use real tokens and users or other
  applications rely on consistent functionality.

When you are getting started, you should use **Testnet.**

## Install tooling

To install Walrus and Sui, use the Mysten Labs `suiup` tool. First,
[install suiup](https://github.com/MystenLabs/suiup?tab=readme-ov-file#installation):

```bash
curl -sSfL https://raw.githubusercontent.com/Mystenlabs/suiup/main/install.sh | sh
```

Then, install `sui` and `walrus`:

```bash
suiup install sui
suiup install walrus
```

### Configure tooling for Walrus Testnet

After you install Walrus, you need to configure the Walrus client, which tells it the RPC URLs to
use to access Testnet or Mainnet. The easiest way to configure Walrus is to download the following
prefilled configuration file.

```bash

> mkdir ~/.config/walrus/

Reviewer: Nit: As we use the `--create-dirs` option in the next line, this is not necessary.

```suggestion
```

> curl --create-dirs https://manansh11.github.io/walrus-docs/setup/client_config_testnet.yaml -o ~/.config/walrus/client_config.yaml

Reviewer: Important:
```suggestion
curl --create-dirs https://docs.wal.app/setup/client_config.yaml -o ~/.config/walrus/client_config.yaml
```

first address.

To confirm the Walrus configuration also uses Testnet, run the command:

```bash
walrus info
```

Make sure that this command's output includes `Epoch duration: 1day` to indicate connection to
Testnet.

For detailed information about the `walrus` command-line tool, use the `walrus --help` command.
Or, append `--help` to any `walrus` subcommand to get details about that specific command.

## Understanding your Sui account

When you ran `sui client` during setup, the system automatically created a Sui account for you.
Sui uses addresses and accounts. When you store blobs on Walrus, Walrus binds them to an object on
Sui that an address owns.

An *address* is a unique location on the blockchain. A 32-byte identifier (displayed as 64 hex
characters with `0x` prefix) identifies the address, which can own objects. The system derives the
address from a public key using a hash function.

Anyone can see addresses, and they are valid on all networks (Testnet, Mainnet, and so on), but
networks do not share data and assets.

An *account* is an address plus the key to access it. If you have an address's private key, you

> can actually use what the address owns, such as tokens and objects.

Reviewer 2: ```suggestion
have privileged access and control over what the address owns, such as tokens and objects.
```

> [Learn more about the Sui faucet.](https://docs.sui.io/guides/developer/getting-started/get-coins)

Now, convert some of those SUI tokens into WAL with the command:

```bash
walrus get-wal --context testnet
```

Then, check your balance again with `sui client balance` to confirm you now have WAL:

```bash
╭─────────────────────────────────────────╮
│ Balance of coins owned by this address  │
├─────────────────────────────────────────┤
│ ╭─────────────────────────────────────╮ │
│ │ coin  balance (raw)     balance     │ │
│ ├─────────────────────────────────────┤ │
│ │ Sui   8869252670        0.05 SUI    │ │
│ │ WAL   500000000         0.50 WAL    │ │
│ ╰─────────────────────────────────────╯ │
╰─────────────────────────────────────────╯
```

## Store a blob

Changes to objects on Sui happen through the use of transactions. Accounts sign these transactions
on behalf of addresses and they result in the system creating, updating, transferring, and
sometimes destroying objects. Learn more about [transactions](https://docs.sui.io/concepts/transactions).

To upload a file to Walrus and store it as a blob, run the following command:

```bash

> walrus store file.txt --epochs 2 --context testnet

Reviewer: Nit: Now that Testnet is set as the main context, the `--context testnet` isn't strictly necessary. However, it also doesn't hurt. 🤷

ID.

## Retrieve a blob

To retrieve a blob and save it on your local machine, run the following command:

```bash
walrus read <blob-id> --out file.txt --context testnet
```

Replace `<blob-id>` with the blob's identifier the `walrus store` command returns in its output,
and replace `file.txt` with the name and file extension for storing the file locally.

## Extend a blob's storage duration

To extend a blob's storage duration, you must reference the Sui object ID and indicate how many
epochs you want to extend the blob's storage for.

Run the following command to extend a blob's storage duration by 3 epochs. You must use the Sui
object ID, not the blob ID:

```bash
walrus extend --blob-obj-id <blob-object-id> --epochs-extended 3 --context testnet
```

Replace `<blob-object-id>` with the blob's Sui object ID the `walrus store` command returns in its
output.

## Delete a blob

To delete a blob, run the following command:

```bash
walrus delete --blob-id <blob-id> --context testnet
```

Replace `<blob-id>` with the blob's identifier the `walrus store` command returns in its output.

## Next steps

> 1. [Build your first Walrus application](https://docs.wal.app/dev-guide/dev-guide.html). Explore

Reviewer: Nit: If we only have a single item, I'd probably remove the numbering.

Reviewer: We could also point to the TS SDK: https://sdk.mystenlabs.com/walrus

> 1. [Build your first Walrus application](https://docs.wal.app/dev-guide/dev-guide.html). Explore

Author: ```suggestion
[Build your first Walrus application](https://docs.wal.app/dev-guide/dev-guide.html). Explore
```

   working examples:
   - [Python examples](https://github.com/MystenLabs/walrus/tree/main/docs/examples/python)
   - [JavaScript web form](https://github.com/MystenLabs/walrus/tree/main/docs/examples/javascript)
   - [Move smart contracts](https://github.com/MystenLabs/walrus/tree/main/docs/examples/move)

## Need help?

> - [Troubleshooting guide](https://docs.wal.app/usage/troubleshooting.html)

Reviewer: Nit: We can reference this with an internal link:

```suggestion
- [Troubleshooting guide](./troubleshooting.html)
```

We could also add the following here or on the troubleshooting page:
```md
If you observe a bug or want to request a feature, please search for an existing
[issue](https://github.com/MystenLabs/walrus/issues) on this topic and, if none exists, create a new
one.
```
